src/study_api.jl

```diff
@@ -0,0 +1,167 @@
+const PSRCLASSES_DEFAULTS = JSON.parsefile(joinpath(@__DIR__, "JSON", "psrclasses.defaults.json"))
+
+function create_element!(
+    data::Data,
+    name::String,
+)
+    if !haskey(PSRCLASSES_DEFAULTS, name)
+        error("Unknown PSR Class '$name'")
+    end
+
+    element = deepcopy(PSRCLASSES_DEFAULTS[name])
+
+    index = _insert_element!(data, name, element)
+
+    return index
+end
+
+function _insert_element!(
+    data::Data,
+    name::String,
+    element::Any,
+)
+    raw_data = _raw(data)
+
+    @assert raw_data isa Dict{String,<:Any}
```

Comment:
These `@assert` macros for type checking seem a bit weird to me. Isn't raw_data always a `Dict{String, <:Any}`. If we want to enforce it to be we could also annotate this as the return type of the `_raw` function. Checking the type of something with assertions in runtime seems a worse design than only allowing it to be a certain type by default